### PR TITLE
Resolve failing test scenarios

### PR DESCRIPTION
**Pull request by MelSumner**
If merged, this PR tries to resolve some scenarios and tidy up some inconsistencies (spelling errors, dependency versions, instructions, etc). 

- resolved differences where node versions were declared
- fixed spelling errors
- removed some dependabot ignores that were no longer relevant
- updated README
- updated CONTRIBUTING
- added some dependencies 
- upgraded some dependencies

Currently, I can't seem to get release/beta/canary to work, but the embroider scenario now passing is an improvement

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,14 +11,14 @@ ember-pikaday provides a datepicker modifier & components for Ember using the Pi
 
 Prerequisites:
 
-- Ember.js v3.25 or above
-- Node.js v12 or above
+- Ember.js v3.28 or above
+- Node.js v20 or above
```
Comment (mansona) [inline README.md:15]: we can remove node prerequisits since this is v2 addon

File: package.json [hunk 2]
```diff
@@ -109,10 +108,10 @@
     }
   },
   "engines": {
-    "node": ">= 20"
+    "node": ">=20.11.0"
   },
```
Comment (mansona) [inline package.json:112]: We should remove the node engine from the package because in the consuming app it doesn't matter what version of node it's using

**Comment by MelSumner** [comment]
@mansona @knownasilya @ef4 I don't know if I can get farther than this without some help, would y'all take a look at the release/beta/canary failures and LMK what you think? TIA!!

File: package.json (hunk 2 above)
Comment (MelSumner) [reply package.json:112]: Should I even leave the "engines" section in there at all then?

**Comment by MelSumner** [comment]
If there's no objections, I'd like to merge this PR, do a release, and then keep working on the release/beta/canary issues so we can at least have some incremental progress.

File: package.json (hunk 2 above)
Comment (mansona) [reply package.json]: no we don't need the engines section any more 👍

**Review by knownasilya — APPROVED** [review]

**Pull request merged**